checker/src/org/checkerframework/checker/index/upperbound/UpperBoundTransfer.java

```diff
@@ -204,6 +205,35 @@ private UBQualifier accountForLowerBoundAnnos(Node node, UBQualifier qual, Strin
         return qual;
     }
 
+    private void propagateToSubterm(
+            Node target, Node offset, UBQualifier qualifier, CFStore store, boolean plusNode) {
+        UBQualifier existingQual =
+                UBQualifier.createUBQualifier(
+                        atypeFactory
+                                .getAnnotatedType(target.getTree())
+                                .getAnnotationInHierarchy(atypeFactory.UNKNOWN));
+
+        UBQualifier newQual =
+                existingQual.glb(
+                        plusNode
+                                ? qualifier.plusOffset(offset, atypeFactory)
+                                : qualifier.minusOffset(offset, atypeFactory));
+
+        Receiver rightRec = FlowExpressions.internalReprOf(analysis.getTypeFactory(), target);
+        store.insertValue(rightRec, atypeFactory.convertUBQualifierToAnnotation(newQual));
+    }
+
+    private void propagateToSubterms(
+            BinaryOperationNode node, UBQualifier qualifier, CFStore store) {
+        boolean plusNode = node instanceof NumericalAdditionNode;
+        // Otherwise this should be a numerical subtraction node.
+
+        propagateToSubterm(
+                node.getRightOperand(), node.getLeftOperand(), qualifier, store, plusNode);
+        propagateToSubterm(
```

Comment:
I tried this out and that's not the case. Consider the following code:

```java
  public void subtraction4(String[] a, @IndexFor("#1") int i) {
        if (1 - i < a.length) {
	    //:: error: (assignment.type.incompatible)
	    @LTLengthOf(value="a", offset="1") int k = i;
        }
    }
```

The expected warning is issued:
```java
ParserOffsetTest.java:51: error: [assignment.type.incompatible] incompatible types in assignment.
	    @LTLengthOf(value="a", offset="1") int k = i;
	                                               ^
  found   : @LTLengthOf(value="a", offset={}) int
  required: @LTLengthOf(value="a", offset="1") int
```